Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/UpdateOptions.cs

```diff
@@ -3,11 +3,40 @@
 
 namespace LoRaWan.Tools.CLI.Options
 {
+    using System;
     using CommandLine;
 
     [Verb("update", HelpText = "Update an existing device in IoT Hub.")]
     public class UpdateOptions
     {
+        public UpdateOptions(AddOptions opts)
```

Comment:
Consider using [Mapperly](https://mapperly.riok.app/) to generate and maintain this so that if new options are added to `AddOptions` and similarly to `UpdateOptions` then this won't have to be kept in sync.

```diff
diff --git a/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/AddOptions.cs b/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/AddOptions.cs
index b2cf28d7..37b330f7 100644
--- a/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/AddOptions.cs
+++ b/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/AddOptions.cs
@@ -6,6 +6,7 @@ namespace LoRaWan.Tools.CLI.Options
     using System;
     using System.Collections.Generic;
     using CommandLine;
+    using Riok.Mapperly.Abstractions;
 
     [Verb("add", HelpText = "Add a new device to IoT Hub.")]
     public class AddOptions
@@ -229,4 +230,10 @@ namespace LoRaWan.Tools.CLI.Options
             )]
         public string Network { get; set; }
     }
+
+    [Mapper]
+    public static partial class AddOptionsMapper
+    {
+        public static partial UpdateOptions ToUpdateOptions(this AddOptions opts);
+    }
 }
diff --git a/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/UpdateOptions.cs b/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/UpdateOptions.cs
index d2f06677..55f073fb 100644
--- a/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/UpdateOptions.cs
+++ b/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/CommandLineOptions/UpdateOptions.cs
@@ -3,40 +3,11 @@
 
 namespace LoRaWan.Tools.CLI.Options
 {
-    using System;
     using CommandLine;
 
     [Verb("update", HelpText = "Update an existing device in IoT Hub.")]
     public class UpdateOptions
     {
-        public UpdateOptions(AddOptions opts)
-        {
-            if (opts == null) throw new ArgumentNullException(nameof(opts));
-            Rx1DrOffset = opts.Rx1DrOffset;
-            AppEui = opts.AppEui;
-            AppSKey = opts.AppSKey;
-            AppKey = opts.AppKey;
-            NwkSKey = opts.NwkSKey;
-            DevEui = opts.DevEui;
-            ABPRelaxMode = opts.ABPRelaxMode;
-            ClassType = opts.ClassType;
-            Deduplication = opts.Deduplication;
-            DownlinkEnabled = opts.DownlinkEnabled;
-            DevAddr = opts.DevAddr;
-            FCntDownStart = opts.FCntDownStart;
-            FCntResetCounter = opts.FCntResetCounter;
-            FCntUpStart = opts.FCntUpStart;
-            KeepAliveTimeout = opts.KeepAliveTimeout;
-            NetId = opts.NetId;
-            NwkSKey = opts.NwkSKey;
-            PreferredWindow = opts.PreferredWindow;
-            Rx1DrOffset = opts.Rx1DrOffset;
-            Rx2DataRate = opts.Rx2DataRate;
-            RxDelay = opts.RxDelay;
-            SensorDecoder = opts.SensorDecoder;
-            Supports32BitFCnt = opts.Supports32BitFCnt;
-        }
-
         [Option(
             "deveui",
             Required = true,
diff --git a/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/LoRaWan.Tools.CLI.csproj b/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/LoRaWan.Tools.CLI.csproj
index 82a27868..8a60272c 100644
--- a/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/LoRaWan.Tools.CLI.csproj
+++ b/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/LoRaWan.Tools.CLI.csproj
@@ -38,6 +38,7 @@
     <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="6.0.1" />
     <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="6.0.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+    <PackageReference Include="Riok.Mapperly" Version="2.7.0" />
   </ItemGroup>
 
   <ItemGroup>
diff --git a/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/Program.cs b/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/Program.cs
index fce29760..a2c7e6eb 100644
--- a/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/Program.cs
+++ b/Tools/Cli-LoRa-Device-Provisioning/LoRaWan.Tools.CLI/Program.cs
@@ -191,7 +191,7 @@ namespace LoRaWan.Tools.CLI
                 }
                 else
                 {
-                    var updateOptions = new UpdateOptions(opts);
+                    var updateOptions = opts.ToUpdateOptions();
                     var newTwins = IoTDeviceHelper.UpdateDeviceTwin(existingDeviceTwins, updateOptions);
                     newTwins.DeviceId = opts.DevEui;
                     isSuccess = await IoTDeviceHelper.WriteDeviceTwin(newTwins, opts.DevEui, configurationHelper, false);
```

Mapperly will generate and maintain the following method implementation:

```c#
#nullable enable
namespace LoRaWan.Tools.CLI.Options
{
    public static partial class AddOptionsMapper
    {
        public static partial LoRaWan.Tools.CLI.Options.UpdateOptions? ToUpdateOptions(this LoRaWan.Tools.CLI.Options.AddOptions? opts)
        {
            if (opts == null)
                return default;
            var target = new LoRaWan.Tools.CLI.Options.UpdateOptions();
            target.DevEui = opts.DevEui;
            target.AppSKey = opts.AppSKey;
            target.NwkSKey = opts.NwkSKey;
            target.DevAddr = opts.DevAddr;
            target.NetId = opts.NetId;
            target.ABPRelaxMode = opts.ABPRelaxMode;
            target.AppEui = opts.AppEui;
            target.AppKey = opts.AppKey;
            target.GatewayId = opts.GatewayId;
            target.SensorDecoder = opts.SensorDecoder;
            target.ClassType = opts.ClassType;
            target.DownlinkEnabled = opts.DownlinkEnabled;
            target.PreferredWindow = opts.PreferredWindow;
            target.Deduplication = opts.Deduplication;
            target.Rx2DataRate = opts.Rx2DataRate;
            target.Rx1DrOffset = opts.Rx1DrOffset;
            target.RxDelay = opts.RxDelay;
            target.KeepAliveTimeout = opts.KeepAliveTimeout;
            target.Supports32BitFCnt = opts.Supports32BitFCnt;
            target.FCntUpStart = opts.FCntUpStart;
            target.FCntDownStart = opts.FCntDownStart;
            target.FCntResetCounter = opts.FCntResetCounter;
            return target;
        }
    }
}
```

Reply:
Addressed with e32dca39f59cd13f21f4afa5256d223b6cfa7cda.

Reply:
Thank you @atifaziz for the help and learning here!